src/map_command_handler.hpp

```diff
@@ -190,10 +190,29 @@ class map_command_handler
 		}
 		else if (help_on_unknown_) {
 			utils::string_map symbols;
-			symbols["command"] = get_cmd();
-			symbols["help_command"] = cmd_prefix_ + "help";
-			print("help", VGETTEXT("Unknown command '$command', try $help_command "
-				"for a list of available commands.", symbols));
+			symbols["prefix"] = cmd_prefix_;
+			if (cmd_prefix_ == "/"){
+				symbols["help_command"] = cmd_prefix_ + "help";
+				symbols["command"] = cmd_prefix_ + get_cmd();
+			}
+			else{
+				symbols["help_command"] = "help";
+				symbols["command"] = get_cmd();
+			}
+			if (get_cmd().find(cmd_prefix_) == 0){
+				if (const command* c = get_command(get_data(0).erase(0,1))) {
```

Comment:
`c` isn't used, so it shouldn't need to store this to a variable.
The `erase(0,1)` is hardcoding the assumed length of `cmd_prefix`
Why use `get_data(0)` instead of `get_cmd()`?